src/components/KSelect/KSelect.vue

```diff
@@ -2,278 +2,195 @@
-          :target="`[id='${selectInputId}']`"
-          :test-mode="!!testMode || undefined"
-          @closed="() => {
-            if (selectedItem && appearance === 'select') {
-              filterStr = selectedItem.label
-            }
-            if (isToggled.value) {
-              toggle()
-            }
-          }"
-          @opened="() => {
-            filterStr = ''
+    <KToggle v-slot="{ toggle, isToggled }">
+      <KPop
+        ref="popperElement"
+        v-bind="boundKPopAttributes"
+        :on-popover-click="() => {
+          toggle()
+          return isToggled.value
+        }"
+        :position-fixed="positionFixed"
+        target=".select-wrapper"
+        @closed="() => {
+          if (selectedItem) {
+            filterQuery = selectedItem.label
+          }
+          if (isToggled.value) {
             toggle()
-            onPopoverOpen()
-          }"
+          }
+        }"
+        @opened="() => {
+          if (enableFiltering) {
+            filterQuery = ''
+          }
+          toggle()
+        }"
+      >
+        <div
+          ref="selectWrapperElement"
+          class="select-wrapper"
+          data-testid="select-wrapper"
+          role="listbox"
+          @click="onSelectWrapperClick"
         >
-          <div
-            v-if="appearance === 'button'"
-            :id="selectInputId"
-            class="k-select-button"
-            data-testid="k-select-input"
-            role="listbox"
-            style="position: relative;"
+          <KInput
+            :id="selectId"
+            v-bind="modifiedAttrs"
+            autocapitalize="off"
+            autocomplete="off"
+            class="select-input"
+            :class="{ 'filtering-disabled': !enableFiltering,
+                      'hide-model-value': hasCustomSelectedItem
+            }"
+            data-testid="select-input"
+            :disabled="isDisabled"
+            :error="error"
+            :help="help"
+            :label="label ? strippedLabel : undefined"
+            :label-attributes="labelAttributes"
+            :model-value="filterQuery"
+            :placeholder="selectedItem && !enableFiltering ? selectedItem.label : placeholderText"
+            :readonly="isReadonly"
+            @blur="onInputBlur"
+            @focus="onInputFocus"
+            @keypress="onInputKeypress"
+            @keyup="(evt: any) => triggerFocus(evt, isToggled)"
+            @update:model-value="onQueryChange"
           >
-            <KButton
-              :id="selectTextId"
-              v-bind="modifiedAttrs"
-              appearance="tertiary"
-              :style="widthStyle"
-              @keyup="(evt: any) => triggerFocus(evt, isToggled)"
+            <template #after>
+              <CloseIcon
+                v-if="isClearVisible"
+                data-testid="clear-selection-icon"
+                role="button"
+                tabindex="0"
+                @click="clearSelection"
+              />
```

Comment:
issue: this clear icon needs a `focus-visible` rule for keyboard nav (box-shadow, etc.)